### PR TITLE
Derive Executable from HasStorage

### PR DESCRIPTION
**Pull request by pmrv**
Implements #726; progress tracked there.

### CODE REVIEW

**Comment by pmrv** [comment]
I'm a bit surprised that the CI works, because the tests fail for me locally.  Can someone double check this?  Otherwise this should fix #726.

**Comment by niklassiemer** [comment]
I just ran the unittests for this PR locally and it ran through :)

**Comment by pmrv** [comment]
@jan-janssen Can you also check whether this resolves #713?

**Comment by jan-janssen** [comment]
> @jan-janssen Can you also check whether this resolves #713?

I guess the point where this fails is if we use a remote cluster which uses different file paths. To address this previously in the database we only stored the version and whether or not and MPIE enabled executable is needed (this is stored as the number of cores is stored 1 = serial, >1 = parallel).

**Comment by pmrv** [comment]
I think as long as you don't explicitly set the executable path, this should not be an issue, since `_executable_select` still builds it from the `executable_lst` and `version`, but I have no way to check.

**Comment by pmrv** [comment]
I guess it could happen that `executable_lst` is written on the submitting host and then read on the running, remote, host.  In that case we just don't want to store it in HDF.

**Comment by pmrv** [comment]
> I think as long as you don't explicitly set the executable path, this should not be an issue, since `_executable_select` still builds it from the `executable_lst` and `version`, but I have no way to check.

Did you have time to think on this @jan-janssen / @Leimeroth ? Otherwise I'll go ahead and merge and we can sort issues out down the line.

**Comment by Leimeroth** [comment]
did you mean to tag @niklassiemer maybe? Or should I check something specifically related to my setup?

**Comment by pmrv** [comment]
> did you mean to tag @niklassiemer maybe? Or should I check something specifically related to my setup?

Yes, since you are actually using remote jobs, I was hoping you could give it a go with this PR.

**Comment by Leimeroth** [comment]
> > did you mean to tag @niklassiemer maybe? Or should I check something specifically related to my setup?
> 
> Yes, since you are actually using remote jobs, I was hoping you could give it a go with this PR.

I tested if the executable version is recovered for a simple vasp job after fetching it back and reload it using pr.load(). The correct version was used on the cluster and recovered correctly from hdf.
Is there something more specific that I should test @pmrv ?

**Comment by pmrv** [comment]
@jan-janssen concern was that the file path might be messed up, but if you could submit and load a finished job, I suppose that works. Thanks!

I'll merge this later today then.

**Comment by niklassiemer** [comment]
Anything holding this back?

**Comment by pmrv** [comment]
> Anything holding this back?

Nah, just forgot about it.

File: pyiron_base/job/generic.py [hunk 1]
```diff
@@ -1626,12 +1631,7 @@ def _type_from_hdf(self):
         Internal helper function to load type and version from HDF5 file root
         """
         self.__obj_type__ = self._hdf5["TYPE"]
-        if self._executable:
-            try:
-                self.executable.version = self._hdf5["VERSION"]
-            except ValueError:
-                self.executable.executable_path = self._hdf5["VERSION"]
-        else:
```
Comment (niklassiemer) [inline pyiron_base/job/generic.py:1634]: @jan-janssen this change might be related with not loading the right version?!

**Pull request merged**